server/test/write.go

```diff
@@ -58,6 +58,196 @@ var writeCommandTests = []writeCommandTest{
 		// output
 		err: serverErrors.DuplicateTupleInWrite(tk),
 	},
+	{
+		_name: "ExecuteWithWriteToIndirectUnionRelationshipReturnsError",
+		// state
+		typeDefinitions: []*openfgapb.TypeDefinition{
+			{
+				Type: "repository",
+				Relations: map[string]*openfgapb.Userset{
+					"writer": {Userset: &openfgapb.Userset_This{}},
+					"owner":  {Userset: &openfgapb.Userset_This{}},
+					"viewer": {
+						Userset: &openfgapb.Userset_Union{
+							Union: &openfgapb.Usersets{
+								Child: []*openfgapb.Userset{
+									{Userset: &openfgapb.Userset_ComputedUserset{
+										ComputedUserset: &openfgapb.ObjectRelation{
+											Object:   "",
+											Relation: "writer",
+										},
+									}},
+									{Userset: &openfgapb.Userset_ComputedUserset{
+										ComputedUserset: &openfgapb.ObjectRelation{
+											Object:   "",
+											Relation: "owner",
+										},
+									}},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		// input
+		request: &openfgapb.WriteRequest{
+			Writes: &openfgapb.TupleKeys{TupleKeys: []*openfgapb.TupleKey{{
+				Object:   "repository:auth0/express-jwt",
```

Comment:
```suggestion
				Object:   "repository:openfga/openfga",
```

Reply:
Hey @rhamzeh, absolutely!  I copied one of the test cases above, so do you want me to change all of those in the test file I'm currently editing in?

Reply:
Ah my apologies - let's keep it as is and consistent with the above, and will raise another issue to tackle replacing later.

Reply:
Issue being tracked [here](https://github.com/openfga/openfga/issues/123) - resolving this comment. Thanks @dblclik!